src/ensembl/src/content/app/entity-viewer/gene-view/components/protein-domain-image/ProteinDomainImage.tsx

```diff
@@ -59,20 +68,33 @@ export const getDomainsByResourceGroups = (
 };
 
 const ProteinDomainImage = (props: ProteinDomainImageProps) => {
-  const { start: transcriptStart } = getFeatureCoordinates(props.transcript);
+  const [data, setData] = useState<Product | null>(null);
 
-  if (!props.transcript.product.protein_domains_resources) {
+  useEffect(() => {
+    fetchTranscript(props.transcriptId).then((result) => {
+      if (result?.product) {
+        setData(result.product);
+      }
+    });
+  }, [props.transcriptId]);
+
+  return data ? <ProteinDomainImageWithData {...props} product={data} /> : null;
```

Comment:
For ProteinDomainImageWithData, `protein` is a better property name than product

Reply:
✔️